### ISSUE TITLE
Labels cut off in chart

### ISSUE THREAD

**Issue by AbeHandler** [masayuki0812/c3#219]
I have a bar chart with rotated axes. The labels are a little long and are getting cut off. Here is a live demo:

https://dl.dropboxusercontent.com/spa/juxddtvfvlf51yv/Sample/test.html

Here is the code: 
var chart = c3.generate({
    data: {
        columns: [
            ['data1', 30, 200, 100, 400, 150, 250, 30, 200, 100, 400, 150, 250, 50, 50, 50, 50]
        ],
        type: 'bar'
    },
    bar: {
        width: {
            ratio: 0.25 // this makes bar width 50% of length between ticks
        }
        // or
        //width: 100 // this makes bar width 100px
    },
    axis: {
        x: {
            type: 'categorized',
            categories: ['City: Police and Fire', 'City: General Fund', 'City: Public Library', 'Board of Liquidation', 'Sewerage and Water Board', 'Audubon Zoo', 'Audubon Aquarium', 'Board of Assessors', 'School Board', 'Levee Board', 'Law Enforcement', 'Economic Development','Parkway and Recreation Department', 'Capital Improvement', 'Street and Traffic Control', 'Police and Fire']
        },
        rotated: true
    }
});

**Comment by ttamminen** [comment]
I think there is something fizzy going on with the padding calculation on mobile devices.

If you go to http://c3js.org/ and in Google Chrome Developer Tools -> Toggle Device Mode -> iPhone 3GS (other will do) and press F5. Y-axis values are cut off.

Any ideas how to fix it?

![mobile_yaxis_cutoff](https://cloud.githubusercontent.com/assets/1038081/6127515/7f28b2c0-b134-11e4-8f62-e84c6dccdb0a.PNG)

**Comment by masayuki0812** [comment]
Sorry, c3js.org is not working correctly. I'm planing the fixed version `0.4.10` shortly, then it should work.

**Comment by nhduc296** [comment]
in demo page this problem still exist

**Comment by aendrew** [comment]
@nhduc296 I just checked — looks fine to me?
![screen shot 2015-07-03 at 15 07 30](https://cloud.githubusercontent.com/assets/185041/8500562/421db9b2-2195-11e5-8533-eaabac3d0809.png)

**Comment by nhduc296** [comment]
@aendrew 

in demo page http://c3js.org/samples/axes_x_tick_format.html

```

var chart = c3.generate({
    data: {
        x: 'x',
        columns: [
            ['x', '2010-01-01', '2011-01-01', '2012-01-01', '2013-01-01', '2014-01-01', '2015-01-01'],
            ['sample', 30, 200, 100, 400, 150, 250]
        ]
    },
    axis : {
        x : {
            type : 'timeseries',
            tick: {
                //format: function (x) { return x.getFullYear(); }
              format: '%d-%m-%Y' // format string is also available for timeseries data
            }
        }
    }
});


```

**Comment by aendrew** [comment]
Ah. Yes indeed, still seems to be an issue with longer titles.

**Comment by eregnier** [comment]
This issue seems still not solved. This is annoying :/ . Anyone how it should be done in a proper way ? Computing padding from total length of labels is still the better solution or is there any better methods ?

**Comment by skrzepij** [comment]
Any update here?

**Comment by GusSalvador** [comment]
I'm still seeing this issue when setting a custom tick count.  The last tick is being cut off.

**Comment by lepolt** [comment]
Having the same issue here. 
![clipped](https://user-images.githubusercontent.com/669410/36222961-e99df6be-1190-11e8-82ca-56c0311e174c.jpg)

**Comment by eregnier** [comment]
@inlinecoder maybe rotating your labels may solve your issue:
http://c3js.org/samples/axes_x_tick_rotate.html

**Comment by inlinecoder** [comment]
Of course, we can find some kind of a workaround, but that doesn't solve the global issue.

Atm, I've managed to overcome overlapping and cropping, at least for now.

What I did is just moved nested styles and made them global. I guess, C3 tries to find matching styles and perform some calculations with it; and if some specific style is nested, C3 can't find it, and basically can't perform a proper calculation.

<img width="1136" alt="screen shot 2018-02-23 at 11 34 33" src="https://user-images.githubusercontent.com/4806743/36587364-adb71622-188d-11e8-8420-85e8d227626c.png">

<img width="552" alt="screen shot 2018-02-23 at 11 35 24" src="https://user-images.githubusercontent.com/4806743/36587441-dfc84424-188d-11e8-92e6-68f616b6971f.png">

_TLDR; Do not nest C3 CSS_

**Comment by mervinva** [comment]
Any update in this issue ? I am still getting this.

**Comment by inlinecoder** [comment]
@mervinva can you attach an exact issue you're having, so it's easier to think how to handle it?

**Comment by mervinva** [comment]
Same issue like the image commented by lepolt above.

**Comment by garthgoodson** [comment]
Same issue for me... Date/Year labels get truncated.

**Comment by victorpinheiro** [comment]
Still having this issue. Any fix?

**Comment by roremeol** [comment]
It woks for me

```css
.c3 .c3-axis-x text:last-child {
  transform: translate(-8px, 0)!important;
}
```

**Comment by inlinecoder** [comment]
@roremeol, that's not a good solution, even if it works.

**Comment by roremeol** [comment]
@inlinecoder, for now until they fix it, it's the best that i have

**Comment by luchnikovs** [comment]
Another way

`
svg {
    overflow: visible !important;
}
`

**Comment by Lampent** [comment]
Having the same problem, anything new?

**Comment by markpalfreeman** [comment]
In the instance of y-axis labels getting cut off, is C3 calculating the width of the outer axis section by the _first_ text length in the data set?

To prevent this, it seems like the library should perform the calculation (similar to @gopeter's [solution above]((similar to @gopeter's solution above))) for the longest label and base the width on that.

**Comment by tedkreutzer96** [comment]
At this point it doesn't seem like @masayuki0812 has any interest in keeping this library updated, 10 years later and the same problem persists lol
